flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStValueState.java

```diff
@@ -87,8 +93,10 @@ public byte[] serializeKey(ContextKey<K, N> contextKey) throws IOException {
                 ctxKey -> {
                     SerializedCompositeKeyBuilder<K> builder = serializedKeyBuilder.get();
                     builder.setKeyAndKeyGroup(ctxKey.getRawKey(), ctxKey.getKeyGroup());
+                    N namespace = contextKey.getNamespace(this);
```

Comment:
Coud we just mark ContextKey#getNamespace is Nonnull, and it will return VoidNamespace if null ?
Then we could avoid introducing a new parameter in the `AsyncKeyedStateBackend`.
Or Will we support other default namespace ?

Reply:
Well we cannot omit `namespace` param and assign a `VoidNamespace.INSTANCE` if not provided, since the `namespaceSerializer` should know how to serialize the `VoidNamespace.INSTANCE` if so.